docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,656 @@
+      authSig: JSON.parse(authSig),
+      chain: chainACondition.chain,
+    });
+  
+    const chainBConditionsPass = await Lit.Actions.checkConditions({
+      conditions: [chainBCondition],
+      authSig: JSON.parse(authSig),
+      chain: chainBCondition.chain,
+    });
+
+    console.log("chainAConditionsPass: ", chainAConditionsPass, "chainBConditionsPass: ", chainBConditionsPass);
+```
+
+Now our action checks if the conditions are passed on both chains, it takes `authSig` as a parameter here and then logs the result of both condition checks. We'll look into auth sigs while we learn to execute lit actions.
+
+For generating signatures inside our Lit Action, we'll use `Lit.Actions.signEcdsa`. This method allows us to use a pkp (which lets an action generate a sign) for generating signatures. Our Lit Action will now decide based on checked conditions and generate appropriately signed transactions which we can send to the blockchain later.
+
+```js
+    if (chainAConditionsPass && chainBConditionsPass) {
+      await generateSwapTransactions();
+      return;
+    }
+
+    if (chainAConditionsPass) {
+      await Lit.Actions.signEcdsa({
+        toSign: hashTransaction(chainAClawbackTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainASignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({
+          chainATransaction: chainAClawbackTransaction,
+        }),
+      });
+      return;
+    }
+  
+    if (chainBConditionsPass) {
+      await Lit.Actions.signEcdsa({
+        toSign: hashTransaction(chainBClawbackTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainBSignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({
+          chainBTransaction: chainBClawbackTransaction,
+        }),
+      });
+      return;
+    }
+
+    const generateSwapTransactions = async () => {
+      await LitActions.signEcdsa({
+        toSign: hashTransaction(chainATransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainASignature",
+      });
+      await LitActions.signEcdsa({
+        toSign: hashTransaction(chainBTransaction),
+        publicKey: pkpPublicKey,
+        sigName: "chainBSignature",
+      });
+      Lit.Actions.setResponse({
+        response: JSON.stringify({ chainATransaction, chainBTransaction }),
+      });
+    };
+```
+
+As everything on the blockchain works on hexadecimal hashes, we need to hash our transactions as well.
+
+```js
+    const hashTransaction = (tx) => {
+      return ethers.utils.arrayify(
+        ethers.utils.keccak256(
+          ethers.utils.arrayify(ethers.utils.serializeTransaction(tx)),
+        ),
+      );
+    }
+
+    Lit.Actions.setResponse({ response: "Conditions for swap not met!" });
+  }
+go();
+```
+
+A must `Lit.Actions.setResponse` needs to be set before we are closing the execution of a Lit Action. This is the response which is shown after the execution of a Lit Action.
```

Comment:
> A must `Lit.Actions.setResponse` needs to be set before we are closing the execution of a Lit Action. This is the response which is shown after the execution of a Lit Action.

"We need to remember to set `Lit.Actions.setResponse` before we finish executing the Lit Action. This defines the response that will be shown when the Lit Action finishes executing"